geodiff/src/geodiffrebase.cpp

```diff
@@ -463,31 +463,46 @@ bool _handle_update( const ChangesetEntry &entry, const RebaseMapping &mapping,
 
   ConflictFeature conflictFeature( pk, entry.table->name );
 
+  bool entryHasChanges = false;
   for ( size_t i = 0; i < numColumns; i++ )
   {
     Value patchedVal = patchedVals[i];
     if ( patchedVal.type() != Value::TypeUndefined && entry.newValues[i].type() != Value::TypeUndefined )
     {
-      // we have edit conflict here: both "old" changeset and the "new" changeset modify the same
-      // column of the same row. Rebased changeset will get the "old" value updated to the new (patched)
-      // value of the older changeset
-      outEntry.oldValues[i] = patchedVal;
-      _addConflictItem( conflictFeature, i, entry.oldValues[i], patchedVal, entry.newValues[i] );
+      if ( patchedVal == entry.newValues[i] )
+      {
+        // both "old" and "new" changeset modify the column's value to the same value - that
+        // means that in our rebased changeset there's no further change and there's no conflict
+        outEntry.oldValues[i].setUndefined();
+        outEntry.newValues[i].setUndefined();
+      }
+      else
+      {
+        // we have edit conflict here: both "old" changeset and the "new" changeset modify the same
+        // column of the same row. Rebased changeset will get the "old" value updated to the new (patched)
+        // value of the older changeset
+        outEntry.oldValues[i] = patchedVal;
+        outEntry.newValues[i] = entry.newValues[i];
+        entryHasChanges = true;
+        _addConflictItem( conflictFeature, i, entry.oldValues[i], patchedVal, entry.newValues[i] );
+      }
     }
     else
     {
-      // otherwise the value is same for both patched and this, so use base value
+      // the "new" changeset stays as is without modifications
       outEntry.oldValues[i] = entry.oldValues[i];
+      outEntry.newValues[i] = entry.newValues[i];
+      // if a column is pkey, it would have "new" value undefined in the entry and that's not an actual change
+      if ( entry.newValues[i].type() != Value::TypeUndefined )
+        entryHasChanges = true;
```

Comment:
Isn't this `else` covering three cases?:
 1. both old and new values are null -> no change
 2. old value is null and new value is not -> change, covered
 3. old value is not null and new value is null -> should result to change, but is not?